### PR TITLE
Terraform module added. SOPS encryption added (GCR KMS)

### PR DESCRIPTION
**Pull request by vitstef**

### CODE REVIEW

File: terraform/deploy-to-gcr/modules/mydrive-gcr-s3/variables.enc.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
...1SUEi8TSTrIQOoLGzHeedRly9OfMsndXDppejapM3YCGhZqwEwkrTt8MrtC5yFzC8TGOw62G0O2Cgjo6bhdQZ3BF/p+7cD2LFtkYVQtWhf8NDbnfk33yXsdMimervTiAkVVieXvGDAWsEskS16/HaUjcal2Biut0eTgzrX1XZVyhcuZpM1yI6Nx5DMdCMUotcuhmlHtdC/i4wJmpfUUKM4ZYVpYLC90e0wuYixTKDdwZLPkDjU2lY+Jzl0MfZ0wYkKZbMoKiFRjTxGwijkoL30a6XXzvX0RHWcCvzlZfwlFsdXpkeRfpwGBPU7v1nzU8U4YG9o+1k98vgdvZ81K/eRTZhJ50mJFzJ50agpUu33lEa5Ridk6gcGZwmtt9g4xn0MKEfCCghLSux0lNGtjZie7hzpvUFikNzMB+KnlIP1uU1lWYVbX1ekL+gRw4TDr5QUyEUxeeW1YundTyFRdpzLQFkV5vfk0d59b0LUwQtqvPtOtJ3KsXaxuF8f34SIY9822a95AuUrD+ZGzRv6bRitDcKmiHl0bKlKZ4lkeCuMhKxI7PpZAKKvaI67KnG5uw2FRe0Ls7Pz5vhUkUBw/bON1uCpr6fggwM9MmcuY1TwFaIV8ekgW7bg2uGYxoKg7Dko5zUeCkAunfJl8I4DMfbzHwCXpx7sH2KtNEQ6saG4UsBIptECnX0xOanc9LFnD/rbBasSFqt2qba80LseJIdv6tY1eHD9FRgmkfKwZ5j1Zcpblb54SSoln0hrEOgimbjYmbwGMAgHAMro+OiM6RGl5zYkMqc6Rl1APe5OJ9AvFvTNj496/1HksvNqqxHNySuDsC9mvuzm8vwvexO7bdueU0xxWirmYu14x8ZuwUzS0CHru7+zyeMOIpyeSS79RnIquc6HFrbhaSFhwhXHdoswSwdzx5c32vikmQK6w6D7KvT0XVK5aJC/5hHuSF6g5u2/xCFNqnod5nvxsheD2M4lEv9ti88jpj+6Fq6Km+dJGQ1EPoRN3Oz32X3tE7qz9ztwy2dOIKxk+ZEZSrygXn6/ViRYcoXU9/bScj4VH+exH87pVdvqfFWpsGeFKtnAXgJAadmqIVKcFfK3Oe8FV/cZ9,iv:OkUydlEPXXNi5vuJzmD91/tfroqAawsTiIPg9Kh3wh4=,tag:HVWpFBHdz5eGPLpOT1YFJw==,type:str]",
```
Comment (aaband) [inline terraform/deploy-to-gcr/modules/mydrive-gcr-s3/variables.enc.tf:2]: would be better to store and encrypt vars as json

File: terraform/deploy-to-gcr/mydrive-gcr-project/terraform.enc.tfstate [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
...4ZdFXeuYPy9Bg7SsrFuICffp2f1/NZApugGYFmSBrd74PAwbb3pys+ruX2FgQ/U9I+vPPniEpHULCtKDM9MVdukot2CS8EDl4reGi6+B4aeMHnjxAkzISL1keI5lgYmqih/5YKjqOpuuigpk0hBOTTHBj+6D8bRtv71tDnqurH+ZC1tdXM7EU3BydJ6Lb0dpV6ytJG37Gy2njfdh1AHrXcaZuOValQQEX74Cz5p06fzIu9oEV+NwtYoxGKjxkECZA0P14FEyPWbcn5fgqUS8AzaW1vsFsPhw+++1hVJU1wY3tYLUvAZ+L4ThNKx0kZURvNvhlEQY+79/7h1ouXyY//6qLWtQ2trNPUVAwuc2hAcaU8xu2q9f4dJ6JTLGOy0NYGN0kVHNFc53f+1R2zjXcw2GmHyEmo8jBIlN9S4/yuguwqfhsSmn8hDePeg790PRUOzFH2c8ZxbWE96FHmrHHLCQmFQSzIlCcf6bc4pweYuzPKEU96DOiksuQCldrsst9JNZLspT8QMo4DhytOeaJV6Lp6YMtTgE3K+fUfxR6oE9d3lCFd3Y9Jwof4Or8kG/mLl2JHY1FVebW9reJdMk/6SpvzlS0b/ffunXK1V90Q51tYEEy21nvtzczUoGcQKTJ8TCmVKy/yDAnfmoe7itwNNLRISAeoQSNEpKDgceUX0JVl/LVE7DPjpKQUUvWJPzyTvPWR3yQf90+lDDy2FXLUz3UspMuFS8PmoU8JvrGS3vignVb+1jc2fxR/6Q/fXmGvKnLxp5F1nSUP7yyPI328dS0MwW+sAPJlNrjx+nfhndGHX8TbTOZ4F6zxy9stzpF6oVjOs/xDa3B6lJB/DMTMVul3ogltT/7E4tfVulHgZsGK0dobZgxiiQ9p6EXNU7bKvhipbz6jPI+t1lEtuBSLvbtH3Qo0n/31ctPOIPXQjnL5oaPNAq47dRYoyWjQToOPQO5IIooY/hAvMVSCzVa50ciMwrO9vEEr/9OR6QEnz6j0piywN/jJRIXUJIl77cYqz6pNTLEauLbLKzwAmSGmFQKICQ9kvkpDEPy7fDkMOajdlY4CyzTBggqCKRpzy9m95fmiX4m95GBJ+LKeZchWB,iv:xbqIhlY0mHOd+YJAEukFL2jd34ka+xHAPFGv7lI0fl4=,tag:k7TGTfzUB6QQf+PJaq+k+w==,type:str]",
```
Comment (aaband) [inline terraform/deploy-to-gcr/mydrive-gcr-project/terraform.enc.tfstate:2]: encrypt --input-type json

File: terraform/deploy-to-gcr/mydrive-gcr-project/main.tf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  project = "just-student-344815"
+  region = "us-central1"
+}
+
+module "mydrive-prod" {
+  source = "../modules/mydrive-gcr-s3"
+}
+
+module "mydrive-dev" {
+  source = "../modules/mydrive-gcr-s3"
```
Comment (aaband) [inline terraform/deploy-to-gcr/mydrive-gcr-project/main.tf:30]: why s3?

**Review by aaband — APPROVED** [review]

File: terraform/deploy-to-gcr/modules/mydrive-gcr-s3/variables.enc.tf (hunk 1 above)
Comment (vitstef) [reply terraform/deploy-to-gcr/modules/mydrive-gcr-s3/variables.enc.tf:2]: need to discuss this

File: terraform/deploy-to-gcr/mydrive-gcr-project/terraform.enc.tfstate (hunk 2 above)
Comment (vitstef) [reply terraform/deploy-to-gcr/mydrive-gcr-project/terraform.enc.tfstate:2]: done in next PR

File: terraform/deploy-to-gcr/mydrive-gcr-project/main.tf (hunk 3 above)
Comment (vitstef) [reply terraform/deploy-to-gcr/mydrive-gcr-project/main.tf:30]: removed -s3 in next PR

**Pull request merged**